lib/util/source-code-fixer.js

```diff
@@ -83,51 +92,43 @@ SourceCodeFixer.applyFixes = function(sourceCode, messages) {
     if (fixes.length) {
         debug("Found fixes to apply");
 
-        // sort in reverse order of occurrence
-        fixes.sort((a, b) => b.fix.range[1] - a.fix.range[1] || b.fix.range[0] - a.fix.range[0]);
+        for (const problem of fixes.sort(compareMessagesByFixRange)) {
+            fix = problem.fix;
+            start = fix.range[0];
+            end = fix.range[1];
 
-        // split into array of characters for easier manipulation
-        const chars = text.split("");
-
-        fixes.forEach(problem => {
-            const fix = problem.fix;
-            let start = fix.range[0];
-            const end = fix.range[1];
-            let insertionText = fix.text;
-
-            if (end < lastFixPos) {
-                if (start < 0) {
-
-                    // Remove BOM.
-                    prefix = "";
-                    start = 0;
-                }
-
-                if (start === 0 && insertionText[0] === BOM) {
-
-                    // Set BOM.
-                    prefix = BOM;
-                    insertionText = insertionText.slice(1);
-                }
-
-                chars.splice(start, end - start, insertionText);
-                lastFixPos = start;
-            } else {
+            // Remain overwrapped fixes as problems.
+            // lastPos's initial value is 0.
+            if (lastPos > 0 && lastPos >= start) {
```

Comment:
I think this will fail to detect a conflict between these two fixes, because there is a special check for `lastPos > 0`:

```js
({ range: [0, 0], text: "foo" });
({ range: [0, 0], text: "bar" });
```

`lastPos` will be 0 after applying the first fix, so `lastPos > 0` will be `false` and the fix will be applied.

Maybe it would be better to set the initial value of `lastPos` to `-Infinity` instead.